mlir/include/mlir/Dialect/Presburger/Parser.h

```diff
@@ -51,51 +57,94 @@ class Token {
     Comma,
     And,
     Or,
+    Arrow,
+    SemiColon,
     Unknown
   };
   Token() : kind(Kind::Unknown) {}
   explicit Token(Kind kind) : kind(kind) {}
   Token(Kind kind, StringRef content) : kind(kind), content(content) {}
 
+  /// Returns true if the token is of Kind kind.
   bool isa(Kind kind);
+
+  /// Retruns the StringRef corresponding to the content.
   StringRef string();
-  static std::string name(Token::Kind kind);
+
+  /// Returns a StringRef with a name for the provided kind.
+  static StringRef name(Token::Kind kind);
 
 private:
   Kind kind;
   StringRef content;
 };
 
+/// Splits the buffer into Tokens and is able to report errors at certain
+/// location while using the provided ErrorCallback.
 class Lexer {
 public:
   Lexer(StringRef buffer, ErrorCallback callback);
 
+  /// Returns the next token without consuming it.
   Token peek();
+
+  /// Returns the next token and consumes it.
   Token next();
+
+  /// Consumes a token with specified kind. If non is present it is a no-op.
   void consume(Token::Kind kind);
+
+  /// Consumes the next token and emits an error if it doesn't match the
+  /// provided kind.
   LogicalResult nextAssertKind(Token::Kind kind);
```

Comment:
Should the name be changed since we now don't assert per se but emit an error?

Reply:
Something like `consumeOrError` or `consumeOrEmitError`?

Reply:
I will go for `consumeKindOrError`